File: code/java/剑指offer57-2.java
class Solution {
    public int[][] findContinuousSequence(int target) {
        List<int[]> res = new ArrayList<int[]>();
        int l=1,r=2;
        while(l<r){
            int sum = (l+r)*(r-l+1)/2;
            if(sum==target){
                int[] temp = new int[r-l+1];
                for(int i=l;i<=r;i++) temp[i-l] = i;
                res.add(temp);
                l++;
            }else if(sum<target) r++;
            else l++;
        }
        return res.toArray(new int[res.size()][]);
    }
}

File: code/python/剑指offer57-2.py
class Solution:
    def findContinuousSequence(self, target: int) -> List[List[int]]:
        l = 1
        r = 2
        res = []
        while l < r:
            s = (l + r) * (r - l + 1) / 2
            if s == target:
                res.append([i for i in range(l, r + 1)])
                l += 1
            elif s > target:
                l += 1
            else:
                r += 1
        return res